core/module_specifier.rs

```diff
@@ -51,174 +48,126 @@ impl fmt::Display for ModuleResolutionError {
   }
 }
 
-#[derive(
-  Debug, Clone, Eq, Hash, PartialEq, serde::Serialize, Ord, PartialOrd,
-)]
 /// Resolved module specifier
-pub struct ModuleSpecifier(Url);
-
-impl ModuleSpecifier {
-  fn is_dummy_specifier(specifier: &str) -> bool {
-    specifier == "<unknown>"
-  }
-
-  pub fn as_url(&self) -> &Url {
-    &self.0
-  }
-
-  pub fn as_str(&self) -> &str {
-    self.0.as_str()
-  }
-
-  /// Resolves module using this algorithm:
-  /// https://html.spec.whatwg.org/multipage/webappapis.html#resolve-a-module-specifier
-  pub fn resolve_import(
-    specifier: &str,
-    base: &str,
-  ) -> Result<ModuleSpecifier, ModuleResolutionError> {
-    let url = match Url::parse(specifier) {
-      // 1. Apply the URL parser to specifier.
-      //    If the result is not failure, return he result.
-      Ok(url) => url,
-
-      // 2. If specifier does not start with the character U+002F SOLIDUS (/),
-      //    the two-character sequence U+002E FULL STOP, U+002F SOLIDUS (./),
-      //    or the three-character sequence U+002E FULL STOP, U+002E FULL STOP,
-      //    U+002F SOLIDUS (../), return failure.
-      Err(ParseError::RelativeUrlWithoutBase)
-        if !(specifier.starts_with('/')
-          || specifier.starts_with("./")
-          || specifier.starts_with("../")) =>
-      {
-        let maybe_referrer = if base.is_empty() {
-          None
-        } else {
-          Some(base.to_string())
-        };
-        return Err(ImportPrefixMissing(specifier.to_string(), maybe_referrer));
-      }
-
-      // 3. Return the result of applying the URL parser to specifier with base
-      //    URL as the base URL.
-      Err(ParseError::RelativeUrlWithoutBase) => {
-        let base = if ModuleSpecifier::is_dummy_specifier(base) {
-          // Handle <unknown> case, happening under e.g. repl.
-          // Use CWD for such case.
-
-          // Forcefully join base to current dir.
-          // Otherwise, later joining in Url would be interpreted in
-          // the parent directory (appending trailing slash does not work)
-          let path = current_dir().unwrap().join(base);
-          Url::from_file_path(path).unwrap()
-        } else {
-          Url::parse(base).map_err(InvalidBaseUrl)?
-        };
-        base.join(&specifier).map_err(InvalidUrl)?
-      }
-
-      // If parsing the specifier as a URL failed for a different reason than
-      // it being relative, always return the original error. We don't want to
-      // return `ImportPrefixMissing` or `InvalidBaseUrl` if the real
-      // problem lies somewhere else.
-      Err(err) => return Err(InvalidUrl(err)),
-    };
-
-    Ok(ModuleSpecifier(url))
-  }
-
-  /// Converts a string representing an absolute URL into a ModuleSpecifier.
-  pub fn resolve_url(
-    url_str: &str,
-  ) -> Result<ModuleSpecifier, ModuleResolutionError> {
-    Url::parse(url_str)
-      .map(ModuleSpecifier)
-      .map_err(ModuleResolutionError::InvalidUrl)
-  }
+pub type ModuleSpecifier = Url;
```

Comment:
This is the main change.

Reply:
If you don't want to have `ModuleSpecifier` being a wrapper for `Url` what's the point of having this alias at all?

Reply:
We can maybe remove it later, but one could argue the semantic difference is valuable.